rllib/policy/policy.py

```diff
@@ -700,27 +715,33 @@ def _get_dummy_batch_from_view_requirements(
                 ret[view_col] = \
                     np.zeros((batch_size, ) + shape[1:], np.float32)
             else:
-                ret[view_col] = np.zeros_like(
-                    [view_req.space.sample() for _ in range(batch_size)])
+                if isinstance(view_req.space, gym.spaces.Space):
+                    ret[view_col] = np.zeros_like(
+                        [view_req.space.sample() for _ in range(batch_size)])
+                else:
+                    ret[view_col] = [view_req.space for _ in range(batch_size)]
+
         return SampleBatch(ret)
 
     def _update_model_inference_view_requirements_from_init_state(self):
-        """Uses this Model's initial state to auto-add necessary ViewReqs.
+        """Uses Model's (or this Policy's) init state to add needed ViewReqs.
```

Comment:
Make this more robust against Policies that don't have a model, but do return something from `get_initial_state()`.